### PR TITLE
Cut dataset starting from N-th image

### PR DESCRIPTION
**Pull request by demid5111**

### CODE REVIEW

File: cut_dataset.py [hunk 1]
```diff
@@ -5,39 +5,43 @@
...
-    parser.add_argument('--output_size', 
-                        type=int, 
+                        help='Full path to the source archive')
+    parser.add_argument('--output_size',
+                        type=int,
+                        required=True,
+                        help='Number of images in the output dataset')
+    parser.add_argument('--first_image',
+                        type=int,
                         required=True,
```
Comment (demid5111) [inline cut_dataset.py]: required??

File: cut_dataset.py [hunk 2]
```diff
@@ -5,39 +5,43 @@
...
     return parser
 
+
 def unarchive(source_archive_dir, output_folder_dir):
     shutil.unpack_archive(source_archive_dir, output_folder_dir)
 
 
-def cut_imagenet(output_size, output_folder_dir):
-    file_names = os.listdir(output_folder_dir)    
+def cut_imagenet(first_image, output_size, output_folder_dir):
```
Comment (demid5111) [inline cut_dataset.py]: 3rd and optional

File: cut_dataset.py [hunk 3]
```diff
@@ -49,32 +53,38 @@ def cut_imagenet(output_size, output_folder_dir):
     if len(image_names) == 0:
         sys.exit('Incorrect dataset format.')
 
+    if first_image >= len(image_names) - output_size:
```
Comment (demid5111) [inline cut_dataset.py]: move to an utility method

File: cut_dataset.py [hunk 4]
```diff
@@ -127,7 +141,7 @@ def cut_voc(output_size, output_folder_dir):
 def archive(new_file_names, source_path, output_archive_name, output_folder_dir, rel_path_finder):
     with tarfile.open(os.path.join(source_path, '{}.tar.gz'.format(output_archive_name)), 'w:gz') as tar:
         for file_name in new_file_names:
-            if os.name=='nt':
+            if os.name == 'nt':
```
Comment (demid5111) [inline cut_dataset.py]: let's consider this again

File: cut_dataset.py [hunk 5]
```diff
@@ -144,17 +158,19 @@ def is_imagenet(dataset_type):
 
 if __name__ == '__main__':
     args = parser().parse_args()
+
     output_folder_dir = os.path.join(args.output_archive_dir, 'temp')
-    output_archive_name = '{}_subset_{}'.format(args.dataset_type, str(args.output_size))
+    output_archive_name = '{}_subset_{}({})'.format(args.dataset_type, str(args.output_size), str(args.first_image))
```
Comment (demid5111) [inline cut_dataset.py]: no brackets

File: cut_dataset.py [hunk 6]
```diff
@@ -5,39 +5,47 @@
...
-                        type=int, 
+                        help='Full path to the source archive')
+    parser.add_argument('--output_size',
+                        type=int,
                         required=True,
-                        help='Number of images in an output dataset')
+                        help='Number of images in the output dataset')
+    parser.add_argument('--first_image',
+                        type=int,
+                        required=False,
```
Comment (aalborov) [inline cut_dataset.py:20]: default

File: cut_dataset.py [hunk 7]
```diff
@@ -5,39 +5,47 @@
...
     return parser
 
+
 def unarchive(source_archive_dir, output_folder_dir):
     shutil.unpack_archive(source_archive_dir, output_folder_dir)
 
 
-def cut_imagenet(output_size, output_folder_dir):
-    file_names = os.listdir(output_folder_dir)    
+def check_subset_size(dataset_size, subset_size, first_image):
```
Comment (aalborov) [inline cut_dataset.py]: is_cut_possible

File: cut_dataset.py [hunk 8]
```diff
@@ -46,52 +54,65 @@ def cut_imagenet(output_size, output_folder_dir):
...
 
     annotation_dir = os.path.join(voc_root_dir, 'Annotations')
     for element in voc_content_root:
         path_to_element = os.path.join(voc_root_dir, element)
         if os.path.isdir(path_to_element) and 'Images' in element:
             images_dir = path_to_element
 
+    images_files = os.listdir(images_dir)[first_image:]
+
+    if check_subset_size(len(images_files), output_size, first_image):
```
Comment (aalborov) [inline cut_dataset.py]: .

File: cut_dataset.py [hunk 9]
```diff
@@ -144,17 +162,22 @@ def is_imagenet(dataset_type):
 
 if __name__ == '__main__':
     args = parser().parse_args()
+
+    if not args.first_image:
+        args.first_image = 0
+
     output_folder_dir = os.path.join(args.output_archive_dir, 'temp')
-    output_archive_name = '{}_subset_{}'.format(args.dataset_type, str(args.output_size))
+    output_archive_name = '{}_subset_{}-{}'.format(args.dataset_type, str(args.first_image), str(args.first_image + args.output_size))
```
Comment (aalborov) [inline cut_dataset.py]: str?

**Pull request merged**